### PR TITLE
Add Deno hosting documentation

### LINKED ISSUE #1889: [Community Toolkit] Deno
### Proposed topic or title

Deno

### Location in table of contents.

integrations/community toolkit/deno

### Reason for the article

We've had a contribution for a Deno hosting integration for Community Toolkit.

### Article abstract

Overview of using Deno hosting integration.

### Relevant searches

_No response_

### PR DESCRIPTION
**Pull request by brian-guerrero**
## Summary

Adds documentation for .NET Aspire Community Toolkit Deno package.

Fixes #1889 

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/community-toolkit/hosting-deno.md](https://github.com/dotnet/docs-aspire/blob/690dac52ddc93a225122f1b96516e1cc6db75174/docs/community-toolkit/hosting-deno.md) | [.NET Aspire Community Toolkit Deno hosting integration](https://review.learn.microsoft.com/en-us/dotnet/aspire/community-toolkit/hosting-deno?branch=pr-en-us-1896) |
| [docs/fundamentals/integrations-overview.md](https://github.com/dotnet/docs-aspire/blob/690dac52ddc93a225122f1b96516e1cc6db75174/docs/fundamentals/integrations-overview.md) | [.NET Aspire integrations overview](https://review.learn.microsoft.com/en-us/dotnet/aspire/fundamentals/integrations-overview?branch=pr-en-us-1896) |
| [docs/toc.yml](https://github.com/dotnet/docs-aspire/blob/690dac52ddc93a225122f1b96516e1cc6db75174/docs/toc.yml) | [docs/toc](https://review.learn.microsoft.com/en-us/dotnet/aspire/toc?branch=pr-en-us-1896) |

### CODE REVIEW

File: docs/community-toolkit/hosting-deno.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
+---
+title: Deno
+description: Learn about the .NET Aspire Community Toolkit Deno hosting extensions package which provides functionality to run Deno applications and tasks.
+ms.date: 10/25/2024
+---
+
+# .NET Aspire Community Toolkit Deno Hosting
```
Comment (IEvangelist) [inline docs/community-toolkit/hosting-deno.md:7]:
```suggestion
# .NET Aspire Community Toolkit Deno hosting integration
```

File: docs/community-toolkit/hosting-deno.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
+---
+title: Deno
```
Comment (IEvangelist) [inline docs/community-toolkit/hosting-deno.md:2]:
```suggestion
title: .NET Aspire Community Toolkit Deno hosting integration
```

File: docs/community-toolkit/hosting-deno.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    .WithEndpoint();
+
+builder.Build().Run();
+```
+
+---
+
+### Run Deno tasks
+
+This integration extension adds support for running tasks that are either specified in a `package.json` or `deno.json`.
```
Comment (IEvangelist) [inline docs/community-toolkit/hosting-deno.md:62]:
```suggestion
This integration extension adds support for running tasks that are either specified in a _package.json_ or _deno.json_.
```

File: docs/community-toolkit/hosting-deno.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+builder.AddDenoApp("oak-demo", "main.ts", permissionFlags: ["-E", "--allow-net"])
+    .WithHttpEndpoint(env: "PORT")
+    .WithEndpoint();
+
+builder.Build().Run();
+```
+
+---
+
```
Comment (IEvangelist) [inline docs/community-toolkit/hosting-deno.md:59]:
```suggestion

```

File: docs/community-toolkit/hosting-deno.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+builder.AddDenoTask("vite-demo", taskName: "dev")
+    .WithHttpEndpoint(env: "PORT")
+    .WithEndpoint();
+
+builder.Build().Run();
+```
+
+---
+
```
Comment (IEvangelist) [inline docs/community-toolkit/hosting-deno.md:75]:
```suggestion

```

File: docs/community-toolkit/hosting-deno.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+### Deno Package Installation
```
Comment (IEvangelist) [inline docs/community-toolkit/hosting-deno.md:76]:
```suggestion
### Deno package installation
```

File: docs/community-toolkit/hosting-deno.md [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+This integration extension adds support for installing dependencies that utilizes `deno install` behind the scenes by simply using
+`WithDenoPackageInstallation`. **This only works with a `deno.lock` file present.**
+
+# [WithDenoPackageInstallation()](#tab/WithDenoPackageInstallation)
+
```
Comment (IEvangelist) [inline docs/community-toolkit/hosting-deno.md:82]:
```suggestion

```

File: docs/community-toolkit/hosting-deno.md [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+var builder = DistributedApplication.CreateBuilder(args);
+
+builder.AddDenoTask("vite-demo", taskName: "dev")
+    .WithDenoPackageInstallation()
+    .WithHttpEndpoint(env: "PORT")
+    .WithEndpoint();
+```
+
+---
+
```
Comment (IEvangelist) [inline docs/community-toolkit/hosting-deno.md:93]:
```suggestion

```

File: docs/community-toolkit/hosting-deno.md [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+builder.Build().Run();
+```
+
+---
+
+### Deno Package Installation
+
+This integration extension adds support for installing dependencies that utilizes `deno install` behind the scenes by simply using
+`WithDenoPackageInstallation`. **This only works with a `deno.lock` file present.**
```
Comment (IEvangelist) [inline docs/community-toolkit/hosting-deno.md:79]:
```suggestion
This integration extension adds support for installing dependencies that utilizes `deno install` behind the scenes by simply using
`WithDenoPackageInstallation`.

> [!NOTE]
> This API only works when a _deno.lock_ file present.
```

**Comment by IEvangelist** [comment]
The other thing that I noticed is that this is missing from the _intergrations-overview.md_ page, there's a section that shows the **Available integrations** and part of that is a subsection with the Community toolkit integrations.

File: docs/community-toolkit/hosting-deno.md [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+The following sections detail various usages, from running Vite applications to using specific package managers.
+
+### Run Deno apps
+
+This integration extension adds support for running a Deno application defined in a script. Since [Deno is secure by default](https://docs.deno.com/runtime/fundamentals/security), permission flags must be specified in `permissionFlags` argument of `AddDenoApp`.
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+builder.AddDenoApp("oak-demo", "main.ts", permissionFlags: ["-E", "--allow-net"])
```
Comment (drewnoakes) [inline docs/community-toolkit/hosting-deno.md:51]: I don't know Deno very well, so maybe this is idiomatic as is, but as a newcomer I'd find `--allow-env` more helpful than `-E`.

They're equivalent according to the docs: https://docs.deno.com/runtime/fundamentals/security/#environment-variables

Comment (brian-guerrero) [reply docs/community-toolkit/hosting-deno.md]: What do you think of me showing both the long form and short form flags?

Comment (IEvangelist) [reply docs/community-toolkit/hosting-deno.md]: I saw 0433682 - but I'm not a fan of both examples in the code like that. I'd instead include text that says that the preceding code demonstrates passing the `permissionFlags` with their fully qualified switches, and that you could just as easily use the short-hand aliases and then list them there.

**Review by IEvangelist — CHANGES_REQUESTED** [review]
Please break out the short form args into part of the description.

File: docs/community-toolkit/hosting-deno.md [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+var builder = DistributedApplication.CreateBuilder(args);
+
+builder.AddDenoApp("oak-demo", "main.ts", permissionFlags: ["--allow-env", "--allow-net"])
+    .WithHttpEndpoint(env: "PORT")
+    .WithEndpoint();
+
+builder.Build().Run();
+```
+
+*The preceding code uses the fully qualified switches. Short hand aliases can be used as well.*
```
Comment (IEvangelist) [inline docs/community-toolkit/hosting-deno.md:58]:
```suggestion
The preceding code uses the fully qualified switches. Alternatively, you can use the equivalent alias as well. For more information, see [Deno docs: Security and permissions](https://docs.deno.com/runtime/fundamentals/security/#permissions).
```

**Review by IEvangelist — APPROVED** [review]

**Pull request merged**